app/src/main/java/org/sopt/careerly_android/ui/viewmodel/HomeViewModel.kt

```diff
@@ -1,27 +1,70 @@
 package org.sopt.careerly_android.ui.viewmodel
 
-import android.app.Activity
-import android.content.Intent
-import android.widget.Button
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
-import org.sopt.careerly_android.ui.home.MultiData
+import org.sopt.careerly_android.domain.PostRepository
+import org.sopt.careerly_android.ui.home.PostsData
+import org.sopt.careerly_android.ui.home.ProfileData
 import org.sopt.careerly_android.ui.viewmodel.MainViewModel.Companion.EXAMPLE_PROFILE_IMAGE
 import javax.inject.Inject
 
-class HomeViewModel @Inject constructor() : ViewModel() {
-    private val _multiData = MutableLiveData<List<MultiData>>()
-    val multiData: LiveData<List<MultiData>>
-        get() = _multiData
+@HiltViewModel
+class HomeViewModel @Inject constructor(private val postRepository: PostRepository) : ViewModel() {
+    val postsData = MutableLiveData<List<PostsData>>()
+    val profileData = MutableLiveData<List<ProfileData>>()
 
-    fun getMultiData() {
-        val data = mutableListOf<MultiData>()
-        for (i in 1..7) {
-            data.add(MultiData("Hello${i}", EXAMPLE_PROFILE_IMAGE, EXAMPLE_PROFILE_IMAGE))
+    fun getProfileData() {
+        viewModelScope.launch {
+            postRepository.getPostList(
+
+            ).onSuccess {
+                val data = mutableListOf<ProfileData>()
+                for (i in 0 until it.hotProfiles.size) {
+                    val item = it.hotProfiles[i]
+                    data.add(
+                        ProfileData(
+                            item.followers,
+                            item.job,
+                            item.userName
+                        )
+                    )
+                }
```

Comment:
data랑 it.hotProfiles랑 데이터 클래스가 달라서 이렇게 구현해놓은 건지 뭔지 자세히 모르겠지만,

```suggestion
                data = it.hotProfiles.map {
                    ProfileData(
                            it.followers,
                            it.job,
                            it.userName
                    )
                }
```
이렇게 collection API인 map을 사용하면 보다 간결하게 표현할 수 있어요 :)

Reply:
아하 좋네요~~